src/pymp4/parser.py

```diff
@@ -776,11 +775,11 @@ def sizeof(self, context=None, **kw):
         return 0
 
 
-Box = PrefixedIncludingSize(Int32ub, Struct(
-    "offset" / TellMinusSizeOf(Int32ub),
-    "type" / Peek(PaddedString(4, "utf8")),
-    Embedded(Switch(this.type, {
-        b"ftyp": FileTypeBox,
+Box = Prefixed(Int32ub, Struct(
+    "offset" / Tell,
+    "type" / Peek(PaddedString(4, "ascii")),
+    "box_body" / Switch(this.type, {
```

Comment:
why "box_body"? Just leave as-is instead

Reply:
After 2 years I can´t remember much of this :) but I remember this being one of the main point. Construct having removed "embedded", it seemed to me the only way to do that would be to create a named "box_body" but of course it's very annoying to use and unnecessary. What would you suggest to avoid this exactly?